dashboard/app/controllers/projects_controller.rb

```diff
@@ -54,10 +54,7 @@ def angular
   end
 
   def load
-    if STANDALONE_PROJECTS[params[:key]][:login_required]
-      authenticate_user!
-    end
-    return if redirect_unless_account_type_requirements_met
+    authorize! :load_project, params[:key]
```

Comment:
If you are using this for multiple methods, you can define it at the top of the class file like

authorize! :load_project, params[:key] except: [ones where you don't use it]